pallets/api/src/fungibles/tests.rs

```diff
@@ -35,21 +70,32 @@ fn transfer_from_works() {
 	new_test_ext().execute_with(|| {
 		let value: Balance = 100 * UNIT;
 		let token = TOKEN;
-		let from = Some(ALICE);
-		let to = Some(BOB);
-
-		// Approve CHARLIE to transfer up to `value` to BOB.
-		create_token_mint_and_approve(ALICE, token, ALICE, value * 2, CHARLIE, value);
+		let from = ALICE;
+		let to = BOB;
+		let spender = CHARLIE;
+
+		// Check error works for `Assets::transfer_approved()`.
+		assert_noop!(
+			Fungibles::transfer_from(signed(spender), token, from, to, value),
+			AssetsError::Unknown
+		);
+		// Approve `spender` to transfer up to `value`.
+		pallet_assets_create_mint_and_approve(spender, token, from, value * 2, spender, value);
+		for origin in vec![root(), none()] {
+			assert_noop!(Fungibles::transfer_from(origin, token, from, to, value), BadOrigin);
+		}
 		// Successfully call transfer from.
-		let alice_balance_before_transfer = Assets::balance(token, &ALICE);
-		let bob_balance_before_transfer = Assets::balance(token, &BOB);
-		assert_ok!(Fungibles::transfer_from(signed(CHARLIE), token, ALICE, BOB, value));
-		let alice_balance_after_transfer = Assets::balance(token, &ALICE);
-		let bob_balance_after_transfer = Assets::balance(token, &BOB);
-		// Check that BOB receives the `value` and ALICE `amount` is spent successfully by CHARLIE.
+		let alice_balance_before_transfer = Assets::balance(token, &from);
+		let bob_balance_before_transfer = Assets::balance(token, &to);
+		assert_ok!(Fungibles::transfer_from(signed(spender), token, from, to, value));
+		let alice_balance_after_transfer = Assets::balance(token, &from);
+		let bob_balance_after_transfer = Assets::balance(token, &to);
```

Comment:
```suggestion
		let to_balance_after_transfer = Assets::balance(token, &to);
```

Please check rest of tests too.